pydocstringformatter/run.py

```diff
@@ -123,11 +113,117 @@ def get_enabled_formatters(self) -> dict[str, _formatting.Formatter]:
 
         return enabled
 
-    def format_files(self, filepaths: list[Path]) -> bool:
-        """Format a list of files."""
+    def format_file_tokens(
+        self, tokens: list[tokenize.TokenInfo], filename: Path
+    ) -> tuple[list[tokenize.TokenInfo], bool]:
+        """Format a list of tokens.
+
+        tokens: List of tokens to format.
+        filename: Name of the file the tokens are from.
+
+        Returns:
+            A tuple containing [1] the formatted tokens in a list
+            and a boolean indicating if the tokens were changed.
+
+        Raises:
+            RuntimeError:
+                If the formatters are not able to get to a stable
+                result after 3 iterations.
+
+                It reports what formatters are still modifying the tokens.
+        """
+        formatted_tokens: list[tokenize.TokenInfo] = []
+
         is_changed = False
 
-        for file in filepaths:
-            is_changed = self.format_file(file) or is_changed
+        for index, tokeninfo in enumerate(tokens):
+            new_tokeninfo = tokeninfo
 
-        return is_changed
+            if _utils.is_docstring(new_tokeninfo, tokens[index - 1]):
+                new_tokeninfo, changers, stable = self.apply_formatters_twice(
+                    new_tokeninfo
+                )
+                is_changed = is_changed or bool(changers)
+
+                # Try again to get a stable result, skipping the formatters that
+                # didn't change the tokens in former passes
+                new_tokeninfo, changers, stable = self.apply_formatters_twice(
```

Comment:
We should only apply once here.

Reply:
It is done internally if no changes happen in the first pass. in "apply_formatters_twice". The name might be slightly missleading but the name "apply_formatters_once_or_twice_if_unstable" was not that appealing

Reply:
OK